### PR TITLE
[sp] add data source transformer templates

### PR DESCRIPTION
**Pull request by shrey-mage**
# Summary
- refactor data source menu items into builder util function
- add data source menu items to "add transformer block" dropdown

# Tests
image

(Waiting on backend bugfix from Sachchit, will update w/ live test then)

cc: @johnson-mage @skunichetty @wangxiaoyou1993

### CODE REVIEW

**Comment by wangxiaoyou1993** [comment]
Let's move "Generic (no template)" above the "Data sources"

**Comment by johnson-mage** [comment]
@shrey-mage Add some padding between the text and right arrow:
image

File: mage_ai/frontend/components/PipelineDetail/AddNewBlocks/index.tsx [hunk 1]
```diff
@@ -7,21 +7,16 @@ import KeyboardShortcutButton from '@oracle/elements/Button/KeyboardShortcutButt
 import Spacing from '@oracle/elements/Spacing';
 import { Add } from '@oracle/icons';
 import { AxisEnum } from '@interfaces/ActionPayloadType';
-import { BlockRequestPayloadType, BlockTypeEnum } from '@interfaces/BlockType';
+import { BlockRequestPayloadType, BlockTypeEnum, BLOCK_TYPE_CONVERTIBLE } from '@interfaces/BlockType';
```
Comment (johnson-mage) [inline mage_ai/frontend/components/PipelineDetail/AddNewBlocks/index.tsx:10]: @shrey-mage missed this in the other PR, but rename `BLOCK_TYPE_CONVERTIBLE` to `CONVERTIBLE_BLOCK_TYPES`.

**Review by johnson-mage — APPROVED** [review]
approved after addressing comments

**Pull request merged**